### PR TITLE
Add nullsong mobskill and made Dahaks able to use it

### PR DESCRIPTION
**Pull request by jmcmorris**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

This implements the missing mobskill, Nullsong, which is used by Dahaks. This implements a new function `countEffectWithFlag`. This is needed since Dahak will only use Nullsong if the there are 3 or more effects to dispel. This work is based off of the data collected by Jimmayus in the [Monster Stats Spreadsheet](https://docs.google.com/spreadsheets/d/1YBoveP-weMdidrirY-vPDzHyxbEI2ryECINlfCnFkLI/edit?gid=57955395#gid=57955395) as well as the [capture done by Melo](https://discord.com/channels/443544205206355968/443894311881408522/1262474594128891957).

## Steps to test these changes

Go fight a dahak and spam `!tp 3000`. It'll never use nullsong. Put some buffs up and then try again and it should eventually use nullsong. Dragons have a lot of mobskills so it can take a while.

### CODE REVIEW

File: scripts/actions/mobskills/nullsong.lua [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+local mobskillObject = {}
+
+mobskillObject.onMobSkillCheck = function(target, mob, skill)
+    if mob:hasStatusEffect(xi.effect.SILENCE) then
+        return 1
+    end
+
+    if target:countEffectWithFlag(xi.effectFlag.DISPELABLE) < 3 then
+        return 1
+    end
```
Comment (Xaver-DaRed) [inline scripts/actions/mobskills/nullsong.lua:17]:
```lua
    if 
        mob:hasStatusEffect(xi.effect.SILENCE) or
        target:countEffectWithFlag(xi.effectFlag.DISPELABLE) < 3
    then
        return 1
    end
```

File: scripts/actions/mobskills/nullsong.lua [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    return 0
+end
+
+mobskillObject.onMobWeaponSkill = function(target, mob, skill)
+    local count = target:dispelAllStatusEffect(xi.effectFlag.DISPELABLE)
+
+    if count == 0 then
+        skill:setMsg(xi.msg.basic.SKILL_NO_EFFECT)
+        return count
+    end
```
Comment (Frankie-hz) [inline scripts/actions/mobskills/nullsong.lua:28]: If the skill cant fire off without having at least 3 effects to dispel is this check ever going to even get hit?

Comment (Xaver-DaRed) [reply scripts/actions/mobskills/nullsong.lua:28]: Its an AoE. Other targets (non-main target) may not have any buffs

Comment (Frankie-hz) [reply scripts/actions/mobskills/nullsong.lua:28]: Ah you are right, missed the AOE part

**Review by Xaver-DaRed — APPROVED** [review]

**Review by zach2good — APPROVED** [review]

**Pull request merged**